### ISSUE TITLE
'Whisper' object has no attribute 'transcribe_audio'

### ISSUE THREAD

**Issue by herbiel** [YuanGongND/whisper-at#16]
File "/opt/whisper/whisper-at/src/noise_robust_asr/intermediate_feat_extract/as_full/extract_as_full_whisper_all.py", line 35, in extract_audio
    _, audio_rep = mdl.transcribe_audio(wav)
  File "/root/miniconda3/envs/whisper-at-new/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1695, in __getattr__
    raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'Whisper' object has no attribute 'transcribe_audio'

**Comment by YuanGongND** [comment]
hi there,

You would need some engineering. See https://github.com/YuanGongND/whisper-at/tree/main#11-whisper-feature-extraction for the instruction.

-Yuan

**Comment by herbiel** [comment]
yes, but i can't find any transcribe_audio function in the whisper or whisper-at

**Comment by YuanGongND** [comment]
please check the instruction

Option 1

> You can use the released version of Whisper-AT and get the audio feature [[at this line]](https://github.com/YuanGongND/whisper-at/blob/d68531414a118b9fcf46d0e1ae9715ee8fa67d0f/package/whisper-at/whisper_at/transcribe.py#L258C25-L258C25), by adding result.audio_features_for_at to the returns of transcribe function. You can get the pooled (x20 temporal downsampling) audio representation of each Whisper audio encoder. E.g., for Whisper-Large, result.audio_features_for_at is of shape [32, 75, 1280], where 32 is the number of Whisper encoder layer, 75 is the time length (30s * 100 = 3000 frames / 2 (Whisper downsample) / 20 (AT downsample) = 75). Note the Whisper window is always 30s. So if your input audio is 10s, you need to slice the first 25 time steps. i.e., [32, 25, 1280]. In addition, with the padded zeros and attention mechanism, the output won't be same as just inputting 10s audio without padding.
> Advantage: polished code.
> Disadvantage: always padded to 30s (which is acceptable), not used in our paper experiments.

for option 1, you need to change the code. change the return of `transcribe` and rename the function to `transcribe_audio`

Option 2

> We also provide our actual code to extract feature at [[here]](https://github.com/YuanGongND/whisper-at/tree/main/src/noise_robust_asr/intermediate_feat_extract). This code is researchy, but trims the audio to 10s instead of padding.
> We don't have time to polish this, use if you are an expert on Whisper.

for option 2, please pip -e install our provided package. 

The transcribe_audio is at https://github.com/YuanGongND/whisper-at/blob/main/src/noise_robust_asr/intermediate_feat_extract/whisper_feat_extracrt/whisper/transcribe.py

-Yuan

**Comment by herbiel** [comment]
yes ,thanks you for you help ,but in the train stage,it still fail ,  File "/opt/whisper/whisper-at/src/whisper_at_train/models.py", line 172, in forward
    audio_rep = audio_rep.reshape(B*self.n_layer, audio_rep.shape[2], audio_rep.shape[3]) # [B*32, 25, 1280]
RuntimeError: shape '[192, 25, 80]' is invalid for input of size 95904000

**Comment by asifsha11** [comment]
> herbiel

hi May i know how did you by pass this error ?

File "/opt/whisper/whisper-at/src/noise_robust_asr/intermediate_feat_extract/as_full/extract_as_full_whisper_all.py", line 35, in extract_audio
_, audio_rep = mdl.transcribe_audio(wav)
File "/root/miniconda3/envs/whisper-at-new/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1695, in getattr
raise AttributeError(f"'{type(self).name}' object has no attribute '{name}'")
AttributeError: 'Whisper' object has no attribute 'transcribe_audio'

Im still facing this issue .

**Comment by herbiel** [comment]
last answer can fix this problem.
for option 2, please pip -e install our provided package.

The transcribe_audio is at https://github.com/YuanGongND/whisper-at/blob/main/src/noise_robust_asr/intermediate_feat_extract/whisper_feat_extracrt/whisper/transcribe.py

**Comment by asifsha11** [comment]
> Reference

can you share the steps to be followed in order ?  I installed the package , But still i have this issue .

**Comment by herbiel** [comment]
just install from the https://github.com/YuanGongND/whisper-at/blob/main/src/noise_robust_asr/intermediate_feat_extract/whisper_feat_extracrt/whisper/ ,it will be normal
